Source/Scene/Cesium3DTileset.js

```diff
@@ -0,0 +1,1654 @@
+     * @param {Boolean} [options.immediatelyLoadDesiredLOD=false] When true, do not progressively refine. Immediately load the desired LOD.
+     * @param {Boolean} [options.loadSiblings=false] Determines whether sibling tiles should be loaded when skipping levels-of-detail. When true, the siblings of any visible and downloaded tile are downloaded as well.
+     *
+     * @exception {DeveloperError} The tileset must be 3D Tiles version 0.0.  See https://github.com/AnalyticalGraphicsInc/3d-tiles#spec-status
+     *
+     * @example
+     * var tileset = scene.primitives.add(new Cesium.Cesium3DTileset({
+     *      url : 'http://localhost:8002/tilesets/Seattle'
+     * }));
+     *
+     * @see {@link https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/README.md|3D Tiles specification}
+     */
+    function Cesium3DTileset(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        var url = options.url;
+
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.string('options.url', url);
+        //>>includeEnd('debug');
+
+        var tilesetUrl;
+        var baseUrl;
+
+        if (getExtensionFromUri(url) === 'json') {
+            tilesetUrl = url;
+            baseUrl = getBaseUri(url, true);
+        } else if (isDataUri(url)) {
+            tilesetUrl = url;
+            baseUrl = '';
+        } else {
+            baseUrl = url;
+            tilesetUrl = joinUrls(baseUrl, 'tileset.json');
+        }
+
+        this._url = url;
+        this._baseUrl = baseUrl;
+        this._tilesetUrl = tilesetUrl;
+        this._root = undefined;
+        this._asset = undefined; // Metadata for the entire tileset
+        this._properties = undefined; // Metadata for per-model/point/etc properties
+        this._geometricError = undefined; // Geometric error when the tree is not rendered at all
+        this._gltfUpAxis = undefined;
+        this._processingQueue = [];
+        this._selectedTiles = [];
+        this._desiredTiles = new ManagedArray();
+        this._selectedTilesToStyle = [];
+        this._loadTimestamp = undefined;
+        this._timeSinceLoad = 0.0;
+
+        var replacementList = new DoublyLinkedList();
+
+        // [head, sentinel) -> tiles that weren't selected this frame and may be replaced
+        // (sentinel, tail] -> tiles that were selected this frame
+        this._replacementList = replacementList; // Tiles with content loaded.  For cache management.
+        this._replacementSentinel  = replacementList.add();
+        this._trimTiles = false;
+
+        this._cullWithChildrenBounds = defaultValue(options.cullWithChildrenBounds, true);
+
+        this._requestHeaps = {};
+        this._hasMixedContent = false;
+
+        this._baseTraversal = new Cesium3DTilesetTraversal.BaseTraversal();
```

Comment:
I agree, we should break these out into standalone files (and also make sure they are tested in isolation properly).